.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,127 @@
+name: E2E Tests
+
+on:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+
+  build:
+    runs-on: buildjet-2vcpu-ubuntu-2004
+    timeout-minutes: 25
+    strategy:
+      matrix:
+        edition: [oss]
+    env:
+      MB_EDITION: ${{ matrix.edition }}
+      INTERACTIVE: false
+    steps:
+    - uses: actions/checkout@v2
+    - name: Prepare front-end environment
+      uses: ./.github/actions/prepare-frontend
+    - name: Prepare back-end environment
+      uses: ./.github/actions/prepare-backend
+
+    - name: Java/AOT prep
+      run: |
+        source "./bin/prep.sh"
+        prep_deps
+
+    - run: ./bin/build version
+    - run: ./bin/build translations
+    - run: ./bin/build frontend
+    - run: ./bin/build licenses
+    - run: ./bin/build drivers
+    - run: ./bin/build uberjar
+
+    - name: Mark with the commit hash
+      run:  git rev-parse --short HEAD > COMMIT-ID
+    - name: Calculate SHA256 checksum
+      run: sha256sum ./target/uberjar/metabase.jar > SHA256.sum
+    - name: Upload JARs as artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: metabase-${{ matrix.edition }}-uberjar
+        path: |
+          ./target/uberjar/metabase.jar
+          ./COMMIT-ID
+          ./SHA256.sum
+
+  e2e-tests:
+    runs-on: buildjet-2vcpu-ubuntu-2004
+    timeout-minutes: 20
+    needs: build
+    name: e2e-tests-${{ matrix.folder }}-${{ matrix.edition }}
+    env:
+      MB_EDITION: ${{ matrix.edition }}
+      DISPLAY: ""
+      QA_DB_ENABLED: true
+      ENTERPRISE_TOKEN: ${{ secrets.ENTERPRISE_TOKEN }}
+    strategy:
+      fail-fast: false
+      matrix:
+        edition: [oss]
+        folder:
+          - "custom-column"
+          - "downloads"
+          - "moderation"
+          - "native"
+          - "smoketest"
+    services:
+      maildev:
+        image: maildev/maildev
+        ports:
+          - "80:80"
+          - "25:25"
+      postgres-sample:
+        image: metabase/qa-databases:postgres-sample-12
+        ports:
+          - "5432:5432"
+      mongo-sample:
+        image: metabase/qa-databases:mongo-sample-4.0
+        ports:
+          - 27017:27017
+      mysql-sample:
+        image: metabase/qa-databases:mysql-sample-8
+        ports:
+          - 3306:3306
+    steps:
+    - uses: actions/checkout@v2
+    - name: Prepare front-end environment
+      uses: ./.github/actions/prepare-frontend
+    - name: Prepare back-end environment
+      uses: ./.github/actions/prepare-backend
+
+    - name: Get Cypress cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/Cypress
+        key: ${{ runner.os }}-Cypress-${{ hashFiles('**/yarn.lock') }}
+    - name: Ensure that Cypress is ready
```

Comment:
This is done already when we run cypress already, right ?

Reply:
The more accurate message should be "Ensure Cypress **executable** is ready" (the binary isn't consistently cached with yarn etc).